Source/WebCore/Modules/model-element/DDModelPlayer.mm

```diff
@@ -336,6 +338,54 @@ void setOpaque(bool opaque)
         client->didUpdate(*this);
 }
 
+bool DDModelPlayer::supportsTransform(TransformationMatrix transformationMatrix)
+{
+    if (m_stageMode != StageModeOperation::None)
+        return false;
+
+    if (RefPtr currentModel = m_currentModel)
+        return currentModel->supportsTransform(transformationMatrix);
+
+    return false;
+}
+
+std::optional<TransformationMatrix> DDModelPlayer::entityTransform() const
+{
+    if (RefPtr model = m_currentModel) {
+        if (auto transform = model->entityTransform())
+            return static_cast<simd_float4x4>(*transform);
+    }
+
+    return std::nullopt;
+}
+
+void DDModelPlayer::setStageMode(StageModeOperation stageMode)
+{
+    m_stageMode = stageMode;
+    if (m_stageMode == StageModeOperation::None)
+        return;
+
+    if (RefPtr model = m_currentModel) {
+        auto [center, extents] = m_currentModel->getCenterAndExtents();
+        simd_float4x4 result = matrix_identity_float4x4;
```

Comment:
Maybe for a follow-up, but I think the ModelProcessPlayer keeps the current rotation in this case.